### ISSUE TITLE
Outline doesn't expand in 6.0a

### ISSUE THREAD

**Issue by Dan-Patterson** [spyder-ide/spyder#21072]
## Problem Description
Just starting out with the new release of 6.0.
module outline doesn't expand even when trying the expand all button and/or restarting spyder.  I may be missing something since I have encountered this before with older versions of spyder.

## Versions

* Spyder version: 6.0.0a1
* Python version: 3.9.16
* Qt version: 5.15.2
* PyQt version: 5.15.9
* Operating System name/version: windows 10 (actually, 11)

### Dependencies

```
# Mandatory:
atomicwrites >=1.2.0                                                        :  1.4.1 (OK)
chardet >=2.0.0                                                             :  4.0.0 (OK)
cloudpickle >=0.5.0                                                         :  2.0.0 (OK)
cookiecutter >=1.6.0                                                        :  2.1.1 (OK)
diff_match_patch >=20181111                                                 :  20230430 (OK)
intervaltree >=3.0.2                                                        :  3.1.0 (OK)
IPython >=7.31.1,<9.0.0,!=8.8.0,!=8.9.0,!=8.10.0,!=8.11.0,!=8.12.0,!=8.12.1 :  7.34.0 (OK)
jedi >=0.17.2,<0.19.0                                                       :  0.18.2 (OK)
jellyfish >=0.7                                                             :  1.0.0 (OK)
jsonschema >=3.2.0                                                          :  4.16.0 (OK)
keyring >=17.0.0                                                            :  23.4.0 (OK)
nbconvert >=4.0                                                             :  7.2.1 (OK)
numpydoc >=0.6.0                                                            :  1.5.0 (OK)
paramiko >=2.4.0                                                            :  3.2.0 (OK)
parso >=0.7.0,<0.9.0                                                        :  0.8.3 (OK)
pexpect >=4.4.0                                                             :  4.8.0 (OK)
pickleshare >=0.4                                                           :  0.7.5 (OK)
psutil >=5.3                                                                :  5.9.0 (OK)
pygments >=2.0                                                              :  2.15.1 (OK)
pylint >=2.5.0,<3.0                                                         :  2.17.4 (OK)
pylint_venv >=3.0.2                                                         :  3.0.2 (OK)
pyls_spyder >=0.4.0                                                         :  0.4.0 (OK)
pylsp >=1.7.2,<1.8.0                                                        :  1.7.3 (OK)
pylsp_black >=1.2.0,<3.0.0                                                  :  1.3.0 (OK)
qdarkstyle >=3.0.2,<3.2.0                                                   :  3.1 (OK)
qstylizer >=0.2.2                                                           :  0.2.2 (OK)
qtawesome >=1.2.1                                                           :  1.2.3 (OK)
qtconsole >=5.4.2,<5.5.0                                                    :  5.4.3 (OK)
qtpy >=2.1.0                                                                :  2.3.1 (OK)
rtree >=0.9.7                                                               :  1.0.1 (OK)
setuptools >=49.6.0                                                         :  65.5.1 (OK)
sphinx >=0.6.6                                                              :  7.0.1 (OK)
spyder_kernels >=3.0.0b1,<3.0.0b2                                           :  3.0.0b1 (OK)
textdistance >=4.2.0                                                        :  4.5.0 (OK)
three_merge >=0.1.1                                                         :  0.1.1 (OK)
watchdog >=0.10.3                                                           :  3.0.0 (OK)
zmq >=22.1.0                                                                :  24.0.1 (OK)

# Optional:
cython >=0.21                                                               :  None (NOK)
matplotlib >=3.0.0                                                          :  3.6.0 (OK)
numpy >=1.7                                                                 :  1.20.1 (OK)
pandas >=1.1.1                                                              :  1.4.4 (OK)
scipy >=0.17.0                                                              :  1.6.2 (OK)
sympy >=0.7.3                                                               :  1.9 (OK)

```

**Comment by dalthviz** [comment]
Hi @Dan-Patterson thank you for the feedback! So are you not able to expand the module outline, right? Do you see the arrow besides the module filename? I was unable to reproduce this:


![image](https://github.com/spyder-ide/spyder/assets/16781833/8c4ea924-b090-4d2b-97fb-27847bfcc8f4)

![image](https://github.com/spyder-ide/spyder/assets/16781833/24b67836-bb03-44f1-a4d2-ade4dbc0c688)

![image](https://github.com/spyder-ide/spyder/assets/16781833/e090da67-caaa-4f59-a373-89516c1f941c)


Maybe is something with the LSP server 🤔. Have you tried to restart the LSP server? For that you can go to the status bar `LSP: Python`. Clicking there a menu should appear with the option to `Restart Python Language Server` 

Any other info in order to better understand what could be happenning is greatly appreciated!

**Comment by Dan-Patterson** [comment]
No arrow beside module names.  LSP restart does nothing, it just spins and does nothing.
I have even been looking at the code to see where events are handled.
I have been comparing versions of packages between 5.4.3 and 6.0.a to try and narrow down potential culprits, but I suspect I am going to have to find see a package list for Windows for 6.0.a that actually works as described.  This issue arose before in a previous 5.x incarnation and it was somewhere in the jedi, jupyter client etc realm that a small dependency fix in  x >= y that narrowed down the fix for me.
Frustrating, because I suspect the culprit isn't in the "required" packages but on one of its dependencies

**Comment by dalthviz** [comment]
> I suspect the culprit isn't in the "required" packages but on one of its dependencies

That could be a quite probable cause :/ Maybe running Spyder on debug mode could give us some pointers about what is failing. To do that you can run something like `spyder --debug-info verbose` .

Besides that, thinking on dependencies, `python-lsp-server` (pylsp) had a [new release a couple of days ago (1.7.4)](https://github.com/python-lsp/python-lsp-server/releases/tag/v1.7.4). Maybe updating to that version could help? 🤔 

Also, from your screenshot, seems like you have the `Show all files` option. The error persists even without that option, right?

Let us know if something of the above helps!

**Comment by Dan-Patterson** [comment]
@dalthviz I updated pylps with no affect.  And yes, I had Show all files selected, but even toggling that option off didn't permit expansion either.
Sadly I work in a restricted environment controlled by other software requirements ( a miniconda with a strict package list where pip and conda-forge are shunned).  The only way to get spyder 5.4.3 as my ide was to install spyder using pip (I know! I know!, not recommended but currently it is the only way).  I tried a pip install of 6.0.0a and the outline expansion fails, whereas it works for a pip install of 5.4.3.
So I am suspecting that it may not only be a small package dependency difference but the pip vs conda differences.
So I am not at a loss for an ide, I will just have to stick to 5.4.3 for coding and I will play with the 6.0.0a environment to narrow down what is causing the issue.
I will close this for now, but keep it in mind since the outline explorer has been an issue in previous versions of spyder.
